tests/e2e/post_upgrade_test.go

```diff
@@ -0,0 +1,185 @@
+
+// This test suite includes test cases to verify basic functionality
+// after upgrade for upgrade test
+
+// +build runpostupgrade
+
+package e2e
+
+import (
+	admincliconst "github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+	dockerconst "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testData  = "Hello World!"
+	testFile  = "hello.txt"
+	testData1 = "Hello World Again!"
+	testFile1 = "hello1.txt"
+	volPath   = dockerconst.ContainerMountPoint
+)
+
+type PostUpgradeTestSuite struct {
+	config         *inputparams.TestConfig
+	esx            string
+	vm1            string
+	vm1Name        string
+	volName1       string
+	volName2       string
+	containerName1 string
+	containerName2 string
+}
+
+func (s *PostUpgradeTestSuite) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping post-upgrade tests")
+	}
+
+	s.esx = s.config.EsxHost
+	s.vm1 = s.config.DockerHosts[0]
+	s.vm1Name = s.config.DockerHostNames[0]
+}
+
+func (s *PostUpgradeTestSuite) SetUpTest(c *C) {
+	s.volName1 = admincliconst.PreUpgradeTestVol
+	s.volName2 = admincliconst.PostUpgradeTestVol
+	s.containerName1 = inputparams.GetUniqueContainerName(c.TestName())
+	s.containerName2 = inputparams.GetUniqueContainerName(c.TestName())
+}
+
+var _ = Suite(&PostUpgradeTestSuite{})
+
+// Test steps:
+// 1. Verify volume created in pre-upgrade test is available
+// 2. Verify the volume is available and detached
+// 3. Attach the volume from a new conatainer
+// 4. Verify volume status is attached
+// 5. Read data from that volume to make sure data written in pre-upgrade test remains same
+// 6. Write more data to that volume
+// 7. Stop the container and verify volume status is detached
+// 8. Start the same container again and read the data from the volume to make sure it matches the
+//    data written in previous step
+// 9. Remove the container and verify volume status is detached
+// 9. Remove the volume created in pre-upgrade test
+// 10. Create a new volume
+// 11. Verify the new volume is available
+// 12. Attach the new volume from a new container
+// 13. Verify the volume status is attached
+// 14. Write some data to the new volume
+// 15. Stop the container and verify the new volume status is detached
+// 16. Start the same container again and read the data from the new volume to make it matches the
+//     data written in previous step
+// 17. Remove the container and verify the new volume status is detached
+// 18. Remove the new volume
+func (s *PostUpgradeTestSuite) TestVolumeLifecycle(c *C) {
+	const (
+		testData  = "Hello World!"
```

Comment:
sorry, why is it again? can you please help me to understand what is the difference in L94-100 & L32-38.